babi.py

```diff
@@ -929,84 +1004,78 @@ def _edit(screen: Screen) -> EditResult:
 
         key = _get_char(screen.stdscr)
 
-        if key.key == curses.KEY_RESIZE:
-            screen.resize()
-        elif key.key in File.DISPATCH:
-            screen.file.DISPATCH[key.key](screen.file, screen.margin)
-        elif key.keyname in File.DISPATCH_KEY:
-            screen.file.DISPATCH_KEY[key.keyname](screen.file, screen.margin)
-        elif key.keyname == b'^K':
-            if prevkey.keyname == b'^K':
-                cut_buffer = screen.cut_buffer
-            else:
-                cut_buffer = ()
-            screen.cut_buffer = screen.file.cut(cut_buffer)
-        elif key.keyname == b'^U':
-            screen.file.uncut(screen.cut_buffer, screen.margin)
-        elif key.keyname == b'M-u':
-            screen.file.undo(screen.status, screen.margin)
-        elif key.keyname == b'M-U':
-            screen.file.redo(screen.status, screen.margin)
-        elif key.keyname == b'^_':
-            response = screen.status.prompt(screen, 'enter line number')
-            if response == '':
-                screen.status.update('cancelled')
-            else:
-                try:
-                    lineno = int(response)
-                except ValueError:
-                    screen.status.update(f'not an integer: {response!r}')
+        for _, value in COMMANDS.items():
```

Comment:
this makes the loop here way slower -- you'd want some sort of mapping from keybind to functions instead of doing a (slightly more than `O(N)` lookup through all of the `COMMANDS` items)

Reply:
This would probably be resolved by your `command.register` suggestion, so that's fair enough

Reply:
This is fixed now.